src/components/button/__next__/button.component.tsx

```diff
@@ -0,0 +1,262 @@
+import React, {
+  forwardRef,
+  ReactNode,
+  useCallback,
+  useImperativeHandle,
+  useRef,
+} from "react";
+import { SpaceProps } from "styled-system";
+
+import StyledButton, {
+  StyledChildContainer,
+  StyledLoadingContainer,
+} from "./button.style";
+import Icon, { IconType } from "../../icon";
+import { LoaderSpinner } from "../../loader-spinner";
+import tagComponent, {
+  TagProps,
+} from "../../../__internal__/utils/helpers/tags/tags";
+import useMediaQuery from "../../../hooks/useMediaQuery";
+
+export type ButtonHandle = {
+  focusButton: () => void;
+} | null;
+
+export interface ButtonProps extends SpaceProps, TagProps {
+  /** Identifies the element(s) offering additional information about the button that the user might require. */
+  "aria-describedby"?: string;
+  /**
+   * The aria-label attribute of the button.
+   * Defaults to the iconType, when the component has only an icon.
+   */
+  "aria-label"?: string;
+  /** Identifies the element(s) labelling the button. */
+  "aria-labelledby"?: string;
+  /** The text that the button displays. */
+  children?: ReactNode;
+  /** Flag to indicate that the button is disabled. */
+  disabled?: boolean;
+  /** Flag to indicate that the button can be full-width. */
+  fullWidth?: boolean;
+  /** The click destination of the button. Using this will transform the button into an anchor. */
+  href?: string;
+  /** Defines the position of the chosen icon relative to the children. */
+  iconPosition?: "left" | "right";
```

Comment:
question: with the composable `children` do we need to have an `iconPosition` prop anymore? If we do should `children` just be a string?